readthedocs/config/config.py

```diff
@@ -167,6 +167,44 @@ def catch_validation_error(self, key):
                 source_position=self.source_position,
             )
 
+    def pop(self, name, container, default, raise_ex):
+        """
+        Search and pop a key inside a dict.
+
+        This will pop the keys recursively if the container is empty.
+
+        :param name: the key name in a dotted form (key.innerkey)
+        :param container: a dictionary that contains the key
+        :param default: default value to return if the key doesn't exists
+        :param raise_ex: if True, raises an exception when a key is not found
+        """
+        key = name[0]
```

Comment:
param name above states "key in dotted form", but this looks to be expecting an array.